starter-analytics-postgres/src/test/groovy/io/micronaut/starter/analytics/postgres/StoreGeneratedProjectStatsSpec.groovy

```diff
@@ -54,51 +55,112 @@ class StoreGeneratedProjectStatsSpec extends Specification implements TestProper
+                    it
+                },
+                new Generated(ApplicationType.DEFAULT, Language.GROOVY, BuildTool.GRADLE, TestFramework.SPOCK, JdkVersion.JDK_11, null).with {
+                    it.setSelectedFeatures([new SelectedFeature("graalvm")])
+                    it
+                },
+                new Generated(ApplicationType.DEFAULT, Language.JAVA, BuildTool.GRADLE, TestFramework.SPOCK, JdkVersion.JDK_11, null).with {
+                    it.setSelectedFeatures([new SelectedFeature("graalvm")])
+                    it
+                }
+        ]
 
         when:
-        HttpStatus status = client.applicationGenerated(generated).get()
+        def status = generated.collect {
+            client.applicationGenerated(it).get()
+        }
 
         then:
-        status == HttpStatus.ACCEPTED
+        status.every { it == HttpStatus.ACCEPTED }
 
         when:
-        def application = repository.list(Pageable.UNPAGED)[0]
+        def applications = repository.list(Pageable.UNPAGED)
 
         then:
-        application.type == generated.type
-        application.language == generated.language
-        application.buildTool == generated.buildTool
-        application.jdkVersion == generated.jdkVersion
-        application.testFramework == generated.testFramework
-        application.features.find { it.name == 'google-cloud-function' }
-        application.micronautVersion == VersionInfo.micronautVersion
-        application.dateCreated
+        with(applications[0]) {
+            type == generated[0].type
+            language == generated[0].language
+            buildTool == generated[0].buildTool
+            jdkVersion == generated[0].jdkVersion
+            testFramework == generated[0].testFramework
+            cloudProvider == generated[0].cloudProvider
+            features.find { it.name == 'google-cloud-function' }
+            micronautVersion == VersionInfo.micronautVersion
+            dateCreated
+        }
+
+        and:
+        with(applications[1]) {
+            type == generated[1].type
+            language == generated[1].language
+            buildTool == generated[1].buildTool
+            jdkVersion == generated[1].jdkVersion
+            testFramework == generated[1].testFramework
+            !cloudProvider
+            features.find { it.name == 'graalvm' }
+            micronautVersion == VersionInfo.micronautVersion
+            dateCreated
+        }
+
+        and:
+        with(applications[2]) {
+            type == generated[2].type
+            language == generated[2].language
+            buildTool == generated[2].buildTool
+            jdkVersion == generated[2].jdkVersion
+            testFramework == generated[2].testFramework
+            !cloudProvider
+            features.find { it.name == 'graalvm' }
+            micronautVersion == VersionInfo.micronautVersion
+            dateCreated
+        }
```

Comment:
this could be transformed into something like:

```groovy
applications.eachWithIndex { app, i ->
    assert app.type == generated[i].type
    //...
}
```

No?

Reply:
Nice, done 👍